include/fi_util.h

```diff
@@ -70,15 +70,13 @@ enum fi_match_type {
 
 /*
  * Provider details
- * TODO: Determine if having this structure (with expanded fields)
- * would help support fi_getinfo.  If so, fill out and update
- * implementation
+ */
 struct util_prov {
-	const struct fi_provider *prov;
-	const struct fi_info	*info;  -- list of provider info's
-	provider specific handlers, e.g. resolve addressing
+	const struct fi_provider	*prov;
+	const struct fi_info		*info;
+	const int			tx_shared_ctx;
+	const int			rx_shared_ctx;
```

Comment:
FWIW, I was okay keeping these as flags, but this is fine.  What I didn't want was flags defined as part of a named enum (e.g. enum fi_foo), since enum fi_foo cannot be used safely.